docker-entrypoint.sh

```diff
@@ -94,11 +92,29 @@ if [ "$1" = '/go-agent/agent.sh' ]; then
       fi
     done
 
+    if [ ! -e "${AGENT_WORK_DIR}/config/agent-bootstrapper-log4j.properties" ]; then
+      try cp -rfv "/go-agent/config/agent-bootstrapper-log4j.properties" "${AGENT_WORK_DIR}/config/agent-bootstrapper-log4j.properties"
+      try chown go:go "${VOLUME_DIR}/config/agent-bootstrapper-log4j.properties"
+    fi
+
+    if [ ! -e "${AGENT_WORK_DIR}/config/agent-launcher-log4j.properties" ]; then
+      try cp -rfv "/go-agent/config/agent-launcher-log4j.properties" "${AGENT_WORK_DIR}/config/agent-launcher-log4j.properties"
+      try chown go:go "${VOLUME_DIR}/config/agent-launcher-log4j.properties"
+    fi
+
+    if [ ! -e "${AGENT_WORK_DIR}/config/agent-log4j.properties" ]; then
+      try cp -rfv "/go-agent/config/agent-log4j.properties" "${AGENT_WORK_DIR}/config/agent-log4j.properties"
+      try chown go:go "${VOLUME_DIR}/config/agent-log4j.properties"
+    fi
+
     setup_autoregister_properties_file "${AGENT_WORK_DIR}/config/autoregister.properties"
-    touch "${STDOUT_LOG_FILE}"
-    chown go:go "${STDOUT_LOG_FILE}"
-    try exec /usr/local/sbin/tini -- /usr/local/sbin/gosu go "$0" "$@" >> ${STDOUT_LOG_FILE} 2>&1
+    try exec /usr/local/sbin/tini -- /usr/local/sbin/gosu go "$0" "$@"
   fi
 fi
 
+# these 3 vars are used by `/go-agent/agent.sh`, so we export
+export AGENT_WORK_DIR
+export AGENT_STARTUP_ARGS="${AGENT_STARTUP_ARGS}${AGENT_STARTUP_ARGS:+ }-Dgo.console.stdout=true"
```

Comment:
You need to use GO_AGENT_SYSTEM_PROPERTIES to pass the system property on to the agent process. In https://github.com/gocd/gocd/blob/master/installers/go-agent/release/agent.sh#L159, AGENT_STARTUP_ARGS is assigned certain values. We don't check if the AGENT_STARTUP_ARGS env var is set or not. GO_AGENT_SYSTEM_PROPERTIES is what we use for this. The args passed to the bootstrapper process and agent process with your change is something like this - 

bootstrapper process =` java -Dgo.console.stdout=true -jar /go-agent/agent-bootstrapper.jar -serverUrl https://10.132.21.251:8154/go`

agent process - `/usr/lib/jvm/java-1.8-openjdk/jre/bin/java-Dcruise.console.publish.interval=10-Xms128m-Xmx256m-Djava.security.egd=file:/dev/./urandom-Dagent.plugins.md5=b51b7cd9acb932f7eebc99d53f03450f-Dagent.binary.md5=r8UQ0htTVuQZMYPqPEbw9Q==-Dagent.launcher.md5=k46i12Ak9Ssifpvf20owVw==-Dagent.tfs.md5=+7noh5lnt6dimQZesKXL0Q==-jaragent.jar-serverUrlhttps://10.132.21.251:8154/go-sslVerificationModeNONE`

As you can see, go.console.stdout is not passed

Reply:
Resolved. Renamed the variable to `GO_AGENT_SYSTEM_PROPERTIES`